### PR TITLE
Loki: Monaco Query Editor enabled by default

### PR DESCRIPTION
**Pull request by matyax**
**What is this feature?**

This PR enables the Loki Monaco Editor by default, by setting the default value of the feature flag to true.

Secondarily, there are some minor improvements to datasource instance generation in related tests.

**Which issue(s) does this PR fix?**:

Part of https://github.com/grafana/grafana/issues/44985

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { AGGREGATION_OPERATORS, RANGE_VEC_FUNCTIONS } from '../../../syntax';
 
 import { CompletionDataProvider } from './CompletionDataProvider';
 import { NeverCaseError } from './NeverCaseError';
-import type { Situation, Label } from './situation';
+import { Situation, Label, getSituation } from './situation';
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts:8]: Here `Situation` is not used.

**Review by ivanahuckova — COMMENTED** [review]
It seems that `public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.test.ts` needs to be refactored. Otherwise looks good! 

Is there a reason that refactor needs to be in this PR? Because otherwise you could split it and do refactor in separate PR.

File: public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts (hunk 1 above)
Comment (matyax) [reply public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts]: That change was not meant for this branch.

**Comment by matyax** [comment]
> Is there a reason that refactor needs to be in this PR? Because otherwise you could split it and do refactor in separate PR.

That commit was meant for another branch! Change undone.

**Review by ivanahuckova — COMMENTED** [review]
Looks good, just 1 thing that was brought up in https://github.com/grafana/grafana/pull/58106 - it would be better to have default value set in the registry, rather than `defaults.ini`.

**Comment by matyax** [comment]
Good catch @ivanahuckova , done

**Review by ivanahuckova — APPROVED** [review]
LGTM, a small nit pick, I think it would make sense to also add `FrontendOnly: true,` to registry, as this is only frontend related feature flag.

**Comment by matyax** [comment]
> I think it would make sense to also add FrontendOnly: true, to registry, as this is only frontend related feature flag.

Good idea.

**Comment by matyax** [comment]
Don't re-review yet because we need to update an e2e suite.

**Comment by matyax** [comment]
I need to check why the updated e2e test is failing in CI.

**Comment by matyax** [comment]
This PR had a failing E2E test, it was approved and with auto merge enabled. So it was merged with a test commit. I will work on a follow up PR to ammend that.

**Pull request merged**